code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -445,17 +490,26 @@ var/list/gaslist_cache = null
 
 	var/moved_moles = 0
 	var/abs_moved_moles = 0
+	var/unique_recieved_gases = FALSE
+	var/unique_sent_gases = FALSE
 
 	//GAS TRANSFER
-	for(var/id in sharer_gases - cached_gases) // create gases not in our cache
-		add_gas(id)
-	for(var/id in cached_gases) // transfer gases
-		if(!sharer_gases[id]) //checking here prevents an uneeded proc call if the check fails.
-			sharer.add_gas(id)
-
+	for(var/id in 1 to GAS_LAST) // create gases not in our cache

```

Comment:
Potentially the most unperformant part of this PR (this is proc/share).

It needs the gasid which you can't get with fastfors. Discuss